src/test/java/uk/gov/ch/oauth/Oauth2Test.java

```diff
@@ -1,55 +1,61 @@
-import static org.mockito.Mockito.verifyNoMoreInteractions;
-import static org.mockito.Mockito.when;
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
-import javax.servlet.http.HttpServletResponse;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Test;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.HttpUrl;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.*;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
+import org.mockito.Spy;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.web.reactive.function.client.ClientResponse;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import okhttp3.HttpUrl;
-import okhttp3.mockwebserver.MockResponse;
-import okhttp3.mockwebserver.MockWebServer;
-import okhttp3.mockwebserver.RecordedRequest;
 import uk.gov.ch.oauth.identity.IIdentityProvider;
+import uk.gov.ch.oauth.nonce.NonceGenerator;
 import uk.gov.ch.oauth.session.SessionFactory;
 import uk.gov.ch.oauth.tokens.OAuthToken;
 import uk.gov.ch.oauth.tokens.UserProfileResponse;
 import uk.gov.companieshouse.session.Session;
 import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.handler.SessionHandler;
 import uk.gov.companieshouse.session.model.SignInInfo;
 import uk.gov.companieshouse.session.store.Store;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
 @ExtendWith(MockitoExtension.class)
 public class Oauth2Test {
 
     public static MockWebServer mockServer;
     private static OAuthToken oauthToken;
     private final SignInInfo signInInfo = new SignInInfo();
+    static final String ORIGINAL_REQUEST_URL = "https://www.example.com?original";
+    static final StringBuffer REQUEST_URL_STRING_BUFFER = new StringBuffer(
+            "https://www.example.com");
+    public static final String AUTHORISE_URI = "https://example.com/authorise";
+    static final String STATE = "state";
 
     @Mock
     public IIdentityProvider identityProvider;
     @Mock
+    private OAuth2StateHandler oAuth2StateHandler;
+    @Mock
+    private NonceGenerator nonceGenerator;
+    @Mock
     public HttpServletResponse httpServletResponse;
     @Mock
     public ClientResponse response;
```

Comment:
These don't appear to be used and I don't think they would be injected.

Reply:
I have removed the unused imports making sure to run the tests to make sure everything is okay.